docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,586 @@
+in the contract, you will receive the following error message:
+```
+
+### `library` annotations
+
+In CVL 2, contract functions declared as library functions must be annotated
+with `library` in the `methods` block.
+
+```{todo}
+If you forget to declare a method as a `library` method, you will receive the
+following error message:
+```
+
+### Required `calldata`, `memory`, or `storage` annotations for reference types
+
+In CVL 2, methods entries for internal functions must contain either `calldata`,
+`memory`, or `storage` annotations for all arguments with reference types (such
+as arrays).
+
+```{todo}
+is `calldata` actually one of the options?
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Summaries only apply to one contract by default
+
+In CVL 1, a summary in the `methods` block applied to all methods with the
+given signature.  Entries that had both an explicit receiver and a summary,
+such as the following, were disallowed:
+
+```cvl
+using C as c
+
+methods {
+    c.f(uint) => NONDET
+}
+```
+
+In CVL 2, summaries only apply to a single contract, unless the old behavior is
+explicitly requested by using `_` as the receiver.  If no contract is specified,
+the default is `currentContract`.
+
+Consider the following example:
+```cvl
+using C as c;
+
+methods {
+    function f(uint)   => NONDET;
+    function c.g(uint) => ALWAYS(4);
+    function h(uint)   => ALWAYS(1);
+    function _.h(uint) => NONDET;
+}
+```
+
+In this example, `currentContract.f` has a `NONDET` summary, `c.g` has an `ALWAYS`
+summary, a call to `currentContact.h` has an `ALWAYS` summary and a call to
+`h(uint)` on any other contract will use a `NONDET` summary.
+
+Summaries for specific contract methods (including the default
+`currentContract`) always override wildcard summaries.
+
+Wildcard entries cannot be declared `optional` or `envfree`, since these
+annotations only make sense for specific contract methods.
+
+```{warning}
+The meaning of your summarizations will change from CVL 1 to CVL 2.  In CVL 2,
+any entry without a `_` will only apply to a single contract!
+```
+
+### Requirements on `returns`
+
+In CVL 2, methods block entries require a `returns` clause in the following
+situations:
+
+ - The method is not a wildcard entry and the contract function returns a value.
+   In this case, the methods block entry must match the contract function's
+   return type.
+
+   ```{todo}
+   Error message
+   ```
+
+ - The method is a wildcard entry that is summarized with a ghost or CVL
+   function summary.  In this case, the return type from the CVL function must
+   be compatible with the declared return type for the method.
+
+   ```{todo}
+   Error message
+   ```
+
+In all other situations, a `returns` clause is forbidden.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+In particular, one cannot specify return types for wildcard entries, as different 
```

Comment:
The posted build of the docs doesn't seem to include this sentence and readthedocs is showing a build failure error https://readthedocs.com/projects/certora-certora-prover-documentation/builds/1382001/